File: src/math_helpers.cpp
#include "math_helpers.h"

#include <QLineF>
#include <QPoint>

#include <math.h>
#include <algorithm>
#include <iostream>

float distance(QPointF *A, QPointF *B){
	return sqrt(pow((B->x() - A->x()), 2) + pow((B->y() - A->y()), 2));
}

QPointF midpoint(QPointF *A, QPointF *B) {
	return QPointF((A->x() + B->x())/2, (A->y() + B->y())/2);
}

bool areCollinear(QPointF *A, QPointF *B, QPointF *C) {
	QLineF AB(*A, *B);
	QLineF BC(*B, *C);
	QLineF AC(*A, *C);
	return ceil(AB.angleTo(BC)) == ceil(AB.angleTo(AC));
}

File: src/math_helpers.py
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *
from PyQt5.QtCore import *

import math

def distance(A, B):
	return math.sqrt(math.pow((B.x() - A.x()), 2) + math.pow((B.y() - A.y()), 2))

def midpoint(A, B):
	return QPointF((A.x() + B.x())/2, (A.y() + B.y())/2)

def slope(A, B):
	return ((B.y() - A.y())/(B.x() - A.x()))

def areCollinear(A, B, C):
	try:
		diff = abs(slope(B,C) - slope(A,B))
	except:
		diff = 1

	return diff <= 0.00000001
